src/Compilers/Test/Core/Metadata/ILValidation.cs

```diff
@@ -320,106 +325,122 @@
-                        }
-                        else if (item.Name == "catchHandler")
-                        {
-                            AddTextual(result, Convert.ToInt32(item.GetAttribute("offset"), 16), "async: catch handler");
-                        }
-                    }
+                    add(Convert.ToInt32(e.Attribute("yield").Value, 16), "async: yield");
+                    add(Convert.ToInt32(e.Attribute("resume").Value, 16), "async: resume");
                 }
+                else if (e.Name == "catchHandler" && e.Parent.Name == "asyncInfo")
+                {
+                    add(Convert.ToInt32(e.Attribute("offset").Value, 16), "async: catch handler");
+                }
+            }
 
-                var sourceLines = source.Split(new[] { "\r\n", "\n", "\r" }, StringSplitOptions.None);
-
-                foreach (XmlNode entry in doc.GetElementsByTagName("sequencePoints"))
+            foreach (var e in methodXml.Descendants())
+            {
+                if (e.Name == "entry" && e.Parent.Name == "sequencePoints")
                 {
-                    foreach (XmlElement item in entry.ChildNodes)
-                    {
-                        AddTextual(result, Convert.ToInt32(item.GetAttribute("offset"), 16), "sequence point: " + SnippetFromSpan(sourceLines, item));
-                    }
+                    var documentId = e.Attribute("document").Value;
+                    
+                    var source = getSource(documentId);
+                    Assert.NotNull(source);
+
+                    add(Convert.ToInt32(e.Attribute("offset").Value, 16), "sequence point: " + SnippetFromSpan(source, e));
                 }
             }
 
             return result;
         }
 
-        private static string SnippetFromSpan(string[] lines, XmlElement span)
+        private static string SnippetFromSpan(SourceText text, XElement sequencePointXml)
         {
-            if (span.GetAttribute("hidden") == "true")
+            if (sequencePointXml.Attribute("hidden")?.Value == "true")
             {
                 return "<hidden>";
             }
 
-            var startLine = Convert.ToInt32(span.GetAttribute("startLine"));
-            var startColumn = Convert.ToInt32(span.GetAttribute("startColumn"));
-            var endLine = Convert.ToInt32(span.GetAttribute("endLine"));
-            var endColumn = Convert.ToInt32(span.GetAttribute("endColumn"));
+            var lines = text.Lines;
+
+            var startLine = Convert.ToInt32(sequencePointXml.Attribute("startLine").Value);
+            var startColumn = Convert.ToInt32(sequencePointXml.Attribute("startColumn").Value);
+            var endLine = Convert.ToInt32(sequencePointXml.Attribute("endLine").Value);
+            var endColumn = Convert.ToInt32(sequencePointXml.Attribute("endColumn").Value);
+
+            var lineSpan = new LinePositionSpan(new(startLine, startColumn), new(endLine, endColumn));
```

Comment:
>new(startLine, startColumn), new(endLine, endColumn) [](http://example.com/codeflow?start=48&length=52)

Sorry but we don't use target-typed new in the compiler codebase, so that we can do textual searches on `new Type`